index.html

```diff
@@ -2138,6 +2139,75 @@ <h2>
                 </li>
               </ol>
             </li>
+            <li>If <var>details</var>["<a>regionCode</a>"] is present and not
+            the empty string:
+              <ol>
+                <li>Let <var>regionCode</var> be the result of <a>strip leading
+                and trailing ASCII whitespace</a> from
+                <var>details</var>["<a>regionCode</a>"] and then
+                  <a data-cite="!INFRA#ascii-uppercase">ASCII uppercasing</a>
+                  the result.
+                </li>
+                <li>
+                  <p>
+                    If <var>regionCode</var> is not a valid <a>country
+                    subdivision code element</a> as per [[!ISO3166-2]]'s
+                    section 5.2 "Structure of country subdivision code
+                    elements" (non-normative details below), throw a
+                    <a>RangeError</a> exception.
+                  </p>
+                  <div class="note" title=
+                  "Structure of country subdivision code elements">
+                    <p>
+                      <strong>Do not implement from this note.</strong> The
+                      structure of a <a>country subdivision code element</a> is
+                      formally defined in [[!ISO3166-2]] (section 5.2).
+                      Although the structure is not expected to change at the
+                      time of writing, implementers are expected to track
+                      updates to [[!ISO3166-2]] directly from ISO.
+                    </p>
+                    <p>
+                      As [[!ISO3166-2]] is not freely available to the general
+                      public, the structure of a <a>country subdivision code
+                      element</a> at the time of publication is as follows:
+                    </p>
+                    <ul>
+                      <li>Two <a>code points</a> that match an [[!ISO3166-1]]
+                      alpha-2 country code.
+                      </li>
+                      <li>A single U+002D (-) <a>code point</a>.
+                      </li>
+                      <li>One, two, or three <a data-cite=
+                      "INFRA#ascii-alphanumeric">ASCII alphanumeric</a> code
+                      points, in any order.
+                      </li>
+                    </ul>
+                  </div>
+                </li>
+                <li>Set <var>address</var>.<a>[[\regionCode]]</a> to
+                <var>regionCode</var>.
+                </li>
+                <li>Let <var>region</var> be the corresponding <a>country
```

Comment:
(Adding comment again, so we can discuss here)

@rsolomakhin, I've tried to summarize what you said about how Chrome does the ordering, while keeping the order in which things are matched optional. However, I added matching on the document's body's language as first, to match what we say in `.show()`. The "Any other criteria the user agent deems suitable" if for the fallback you described, whereby you pick the first available.  

WDYT? Would this work?

Reply:
This works.

Reply:
@mnoorenberghe, wdty?

Reply:
I'm not sure what the specific question is… do you want me to comment on the locale ordering or the spec change in general? For the locale ordering, as long as the UA isn't required to show the PR dialog in a language that the UA doesn't have installed then I'm fine with it.

Reply:
On a separate note, I wonder whether we really need to derive `region` from the `regionCode`… it would be simpler to also return the ISO3166-2 code as the `region`.

One concern I have with the spec change is that I've never seen ISO3166-2 used with the country code prefix in the wild. As a web developer I would expect `regionCode` to be "CA" for California, not "US-CA" if I hadn't read documentation on the API and only saw the webidl FWIW. I'm not sure if there are exceptions to cutting off the first 3 characters in order to get the region code specific to the country. If there is then this will likely cause issues for merchants when mapping the PR's address to existing systems.

Reply:
Coming back to this and re-reading the ISO spec, I can't determine that there would be significant issues with chopping off country and dash. I'm still reluctant to do it tho. I'd rather just stick to the full code "just in case" and just so we are not creating new conventions. @mnoorenberghe, is that ok?

Reply:
Although I'm always hesitant to parse other people's strings, ISO 3166-2 is defined to use ISO 3166-1 alpha-2 code elements as the first two characters of the string, followed by the hyphen character, followed by the region itself. So I _think_ we'd be safe in truncated the first three chars.